### PR TITLE
civetweb: bump openssl

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Comment by SpaceIm** [comment]
```
ERROR: Version range '^1' from requirement 'openssl/[^1]' required by 'civetweb/1.13' could not be resolved in remote 'conan-center'
```

@jcar87 @memsharded Is it not supposed to be a valid version range, equivalent to `>=1 < 2`?

**Comment by SpaceIm** [comment]
```
ERROR: Version range '>=1 <2' from requirement 'openssl/[>=1 <2]' required by 'civetweb/1.15' could not be resolved in remote 'conan-center'
```

?

**Review by jwillikers — DISMISSED** [review]

File: recipes/civetweb/all/conanfile.py [hunk 1]
```diff
@@ -82,7 +82,10 @@ def layout(self):
 
     def requirements(self):
         if self.options.with_ssl:
-            self.requires("openssl/1.1.1t")
+            if Version(self.version) < "1.16":
```
Comment (jcar87) [inline recipes/civetweb/all/conanfile.py:85]:
```
            if Version(self.version) < "1.16":
                self.requires("openssl/1.1.1t")
            else:
                self.requires("openssl/[>=1 <4]")
```

I would suggest something like this - indeed Conan 1.x will not be able to resolve a version range that does not include OpenSSL 3.x - I wouldn't complicate things, if the older versions of civetweb are _not_ compatible with OpenSSL 3.x, we can keep it to the most recent OpenSSL 1.1.1.

OpenSSL 1.1.1 will be on its end of life later this year - https://www.openssl.org/blog/blog/2023/03/28/1.1.1-EOL/ - I think there won't be may more version bumps anyway in this period of a few months until then.

**Review by jcar87 — CHANGES_REQUESTED** [review]
See comment re: requirement on OpenSSL

File: recipes/civetweb/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/civetweb/all/conanfile.py:85]: done

**Review by jwillikers — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]

**Pull request merged**